databaseQueries/solutions.js

```diff
@@ -0,0 +1,160 @@
+/**
+ * name : solutions.js
+ * author : Praveen
+ * created-date : 13-Jun-2024
+ * Description : solutions helper for DB interactions.
+ */
+
+// Dependencies
+
+/**
+ * Solutions
+ * @class
+ */
+
+module.exports = class Solutions {
+  /**
+   * find Solution .
+   * @method
+   * @name solutionDocuments
+   * @param {Array} [filterData = "all"] - solutions filter query.
+   * @param {Array} [fieldsArray = "all"] - projected fields.
+   * @param {Array} [skipFields = "none"] - field not to include
+   * @returns {Array} solutions details.
+   */
+
+  static solutionDocuments(filterData = 'all', fieldsArray = 'all', skipFields = 'none') {
+    return new Promise(async (resolve, reject) => {
+      try {
+        let queryObject = filterData != 'all' ? filterData : {};
+        let projection = {};
+        if (fieldsArray != 'all') {
+          fieldsArray.forEach((field) => {
+            projection[field] = 1;
+          });
+        }
+
+        if (skipFields !== 'none') {
+          skipFields.forEach((field) => {
+            projection[field] = 0;
+          });
+        }
+        let solutionsDoc = await database.models.solutions.find(queryObject, projection).lean();
+
+        return resolve(solutionsDoc);
+      } catch (error) {
+        return reject(error);
+      }
+    });
+  }
+
+  /**
+   * findOne Solution .
+   * @method
+   * @name findOne
+   * @param {Array} [filterData = "all"] - solutions filter query.
+   * @param {Array} [fieldsArray = "all"] - projected fields.
+   * @param {Array} [skipFields = "none"] - field not to include
+   * @returns {Array} solutions details.
+   */
+
+  static findOne(filterData = 'all', fieldsArray) {
+    return new Promise(async (resolve, reject) => {
+      try {
+        let queryObject = filterData != 'all' ? filterData : {};
+        let solutionsDoc = await database.models.solutions.findOne(queryObject, fieldsArray);
+
+        return resolve(solutionsDoc);
+      } catch (error) {
+        return reject(error);
+      }
+    });
+  }
+
+  /**
+   * Update solution document.
+   * @method
+   * @name updateSolutionDocument
+   * @param {Object} query - query to find document
+   * @param {Object} updateObject - fields to update
```

Comment:
@praveenKDass description for return data is missing

Reply:
added @VISHNUDAS-tunerlabs